core/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -222,31 +231,119 @@ private[spark] object Logging {
     val binderClass = StaticLoggerBinder.getSingleton.getLoggerFactoryClassStr
     "org.slf4j.impl.Log4jLoggerFactory".equals(binderClass)
   }
-}
 
-private class SparkShellLoggingFilter extends Filter {
 
-  /**
-   * If sparkShellThresholdLevel is not defined, this filter is a no-op.
-   * If log level of event is not equal to root level, the event is allowed. Otherwise,
-   * the decision is made based on whether the log came from root or some custom configuration
-   * @param loggingEvent
-   * @return decision for accept/deny log event
-   */
-  def decide(loggingEvent: LoggingEvent): Int = {
-    if (Logging.sparkShellThresholdLevel == null) {
-      Filter.NEUTRAL
-    } else if (loggingEvent.getLevel.isGreaterOrEqual(Logging.sparkShellThresholdLevel)) {
-      Filter.NEUTRAL
-    } else {
-      var logger = loggingEvent.getLogger()
-      while (logger.getParent() != null) {
-        if (logger.getLevel != null || logger.getAllAppenders.hasMoreElements) {
-          return Filter.NEUTRAL
+  private class SparkShellLoggingFilter extends Filter {
```

Comment:
I think we could potentially simplify this class by having it extend `AbstractFilter` instead of `Filter`:

From AbstractFilter's Javadoc:

```java
/**
 * Users should extend this class to implement filters. Filters can be either context wide or attached to
 * an appender. A filter may choose to support being called only from the context or only from an appender in
 * which case it will only implement the required method(s). The rest will default to return {@link org.apache.logging.log4j.core.Filter.Result#NEUTRAL}.
 * <p>
 * Garbage-free note: the methods with unrolled varargs by default delegate to the
 * {@link #filter(Logger, Level, Marker, String, Object...) filter method with vararg parameters}.
 * Subclasses that want to be garbage-free should override these methods to implement the appropriate filtering
 * without creating a vararg array.
 * </p>
 */
public abstract class AbstractFilter extends AbstractLifeCycle implements Filter {
```

Reply:
Okay. I will try to rewrite this with `AbstractFilter`. I think the functionality should be the same but yea it looks better to use `AbstractFilter` here.